internal/storage/v2/elasticsearch/tracestore/to_dbmodel.go

```diff
@@ -48,31 +46,66 @@ func ToDBModel(td ptrace.Traces) []dbmodel.Span {
 	return batches
 }
 
-func resourceSpansToDbSpans(resourceSpans ptrace.ResourceSpans) []dbmodel.Span {
+type toDBModel struct {
+	allTagsAsFields   bool
+	tagKeysAsFields   map[string]bool
+	tagDotReplacement string
+}
+
+func newToDBModel(allTagsAsFields bool, tagKeysAsFields []string, tagDotReplacement string) *toDBModel {
+	tags := map[string]bool{}
+	for _, k := range tagKeysAsFields {
+		tags[k] = true
+	}
+	return &toDBModel{
+		allTagsAsFields:   allTagsAsFields,
+		tagKeysAsFields:   tags,
+		tagDotReplacement: tagDotReplacement,
+	}
+}
+
+// splitTagsToNestedAndElevatedTags split the tags into nested and elevated tags by replacing the dot in the keys of elevated tags by tagDotReplacement
```

Comment:
> split ... by replacing the dot

this isn't the main functionality of the method, it's a minor detail.

```suggestion
// splitTagsToNestedAndElevatedTags splits the tags into nested and elevated tags
// according to `allTagsAsFields` and `tagKeysAsFields` settings. It also replaces
// dots in the elevated tag keys with a placeholder.
```